code/lib/builder-webpack5/src/preview/loaders.ts

```diff
@@ -0,0 +1,43 @@
+import { getProjectRoot } from '@storybook/core-common';
+import type { Options } from '@swc/core';
+import type { TypescriptOptions } from '../types';
+
+export const createBabelLoader = (options: any, typescriptOptions: TypescriptOptions) => {
+  return {
+    test: typescriptOptions.skipBabel ? /\.(mjs|jsx?)$/ : /\.(mjs|tsx?|jsx?)$/,
+    use: [
+      {
+        loader: require.resolve('babel-loader'),
+        options,
+      },
+    ],
+    include: [getProjectRoot()],
+    exclude: /node_modules/,
+  };
+};
+
+export const createSWCLoader = (options: any) => {
+  const config: Options = {
+    jsc: {
+      parser: {
+        syntax: 'typescript',
+        tsx: true,
+        dynamicImport: true,
+      },
+    },
+  };
+  return {
+    test: /\.(mjs|cjs|tsx?|jsx?)$/,
+    use: [
+      {
+        loader: require.resolve('swc-loader'),
+        options: {
+          ...config,
+          ...options,
```

Comment:
This way to merge the options will haunt us later. If options specify `jsc`, they will completely override the default `jsc` options from the config above. We need to merge them more intelligently. Or not support custom options at all.

Reply:
Unless we explicitly choose to not let users do that.

Similar to babel config, it should actually really come from swc's config file.

Reply:
So I think the solution here is to remove the possibility for `createSWCLoader` to take in any options at all.

Reply:
done